src/app/shared/post-it-collections/post-it-collections.component.ts

```diff
@@ -82,142 +77,79 @@ export class PostItCollectionsComponent implements OnInit, AfterViewInit, OnDest
 
   private collectionsSubscription: Subscription;
 
-  constructor(private collectionService: CollectionService,
-              private searchService: SearchService,
-              private notificationService: NotificationService,
-              private importService: ImportService,
-              private store: Store<AppState>,
+  constructor(private store: Store<AppState>,
               private zone: NgZone) {
   }
 
-  public ngOnInit(): void {
+  public ngOnInit() {
     this.createLayout();
     this.subscribeOnNavigation();
     this.subscribeOnCollections();
   }
 
-  public ngAfterViewInit(): void {
+  public ngAfterViewInit() {
     this.layout.initialize();
   }
 
+  private createLayout() {
+    const config = new PostItLayoutConfig();
+    config.dragEnabled = false;
+
+    this.layout = new PostItLayout('post-it-collection-layout', config, this.zone);
+  }
+
   private subscribeOnNavigation() {
     this.navigationSubscription = this.store.select(selectNavigation).pipe(
       filter(navigation => Boolean(navigation && navigation.workspace && navigation.workspace.organizationCode && navigation.workspace.projectCode))
     ).subscribe(navigation => {
       this.workspace = navigation.workspace;
       this.query = navigation.query;
 
-      this.store.dispatch(new Get({query: this.query}));
+      this.store.dispatch(new CollectionsAction.Get({query: this.query}));
     });
   }
 
-  private createLayout(): void {
-    const config = new PostItLayoutConfig();
-    config.dragEnabled = false;
-
-    this.layout = new PostItLayout('post-it-collection-layout', config, this.zone);
+  private subscribeOnCollections() {
+    this.collectionsSubscription = this.store.select(selectCollectionsByQuery).subscribe(collections => {
+      this.postIts = collections.map(collection => this.collectionToPostIt(collection))
+        .concat(this.unitializedCollections());
+    });
   }
 
-  private collectionToPostIt(collection: CollectionModel, initialized: boolean): PostItCollectionModel {
+  private collectionToPostIt(collection: CollectionModel): PostItCollectionModel {
     const postIt = new PostItCollectionModel;
     postIt.collection = collection;
-    postIt.initialized = initialized;
 
     return postIt;
   }
 
   public toggleFavorite(collection: CollectionModel) {
-    this.collectionService.toggleCollectionFavorite(CollectionConverter.toDto(collection))
-      .subscribe(success => {
-        if (success) {
-          collection.favourite = !collection.favourite;
-        }
-      });
-  }
-
-  public hasWriteRole(collection: CollectionModel): boolean {
-    return this.hasRole(collection, Role.Write);
-  }
+    const collectionCopy = Object.assign({}, collection);
```

Comment:
+1 on not modifying the object directly. But there is an easier way of making a copy:
```typescript
const collectionCopy = {...collection};
```

Reply:
Does it do a deep copy?

Reply:
No, but `Object.assign()` doesn't do that neither. See [examples section on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).